nifi-frontend/src/main/frontend/apps/update-attribute/src/app/pages/update-attribute/ui/rule-listing/rule-listing.component.html

```diff
@@ -0,0 +1,187 @@
+                    <div class="flex gap-x-1 items-center">
+                        <div>Reorder rules</div>
+                        <mat-slide-toggle
+                            [checked]="allowRuleReordering"
+                            (change)="setAllowRuleReordering($event)"></mat-slide-toggle>
+                    </div>
+                }
+                @if (!newRule) {
+                    <button
+                        mat-icon-button
+                        class="primary-icon-button"
+                        [disabled]="!isEditable || allowRuleReordering"
+                        (click)="populateNewRule()">
+                        <i class="fa fa-plus"></i>
+                    </button>
+                }
+            </div>
+        </div>
+        @if (filteredRulesList.length === 0 && !newRule) {
+            <div class="unset">No matching rules</div>
+        } @else {
+            @if (allowRuleReordering) {
+                <mat-accordion
+                    [multi]="true"
+                    cdkDropList
+                    [cdkDropListDisabled]="!isEditable"
+                    [cdkDropListData]="rulesList"
+                    (cdkDropListDropped)="reorderRules($event)">
+                    @for (rule of filteredRulesList; track rule.id) {
+                        <mat-expansion-panel cdkDrag (opened)="panelOpened(rule)" (closed)="panelClosed(rule)">
+                            <mat-expansion-panel-header
+                                class="reorder-header"
+                                [collapsedHeight]="'42px'"
+                                [expandedHeight]="'42px'">
+                                <mat-panel-title>
+                                    <div class="flex justify-between items-center">
+                                        <span class="grip -mt-0.5 mr-5 cursor-grab"></span>
+                                        <div class="flex-1 truncate">
+                                            {{ rule.name }}
+                                        </div>
+                                    </div>
+                                </mat-panel-title>
+                                <mat-panel-description>
+                                    {{ rule.comments }}
+                                </mat-panel-description>
+                            </mat-expansion-panel-header>
+                            <edit-rule
+                                [id]="rule.id"
+                                [name]="rule.name"
+                                [existingRuleNames]="getExistingRuleNames()"
+                                [comments]="rule.comments"
+                                [conditions]="rule.conditions"
+                                [actions]="rule.actions"
+                                [editable]="false"></edit-rule>
+                        </mat-expansion-panel>
+                    }
+                </mat-accordion>
```

Comment:
The drag and drop styles still don't match here and I still think we should not try to combine the expansion panel UX with the drag/drop UX. 

```suggestion
                <div
                    class="cdk-list cdk-drag-selected border"
                    cdkDropList
                    [cdkDropListDisabled]="!isEditable"
                    [cdkDropListData]="rulesList"
                    (cdkDropListDropped)="reorderRules($event)">
                    @for (rule of filteredRulesList; track rule.id) {
                        <div
                            class="border mat-elevation-z2 m-1 font-bold"
                            cdkDragPreviewContainer="parent"
                            cdkDrag>
                            <div class="flex w-full items-center">
                                <span class="grip -mt-1 pr-5"></span>
                                <div class="truncate" [title]="rule.name">
                                    {{ rule.name }}
                                </div>
                                @if (rule.comments) {
                                    <i
                                        class="pl-2 pr-2 fa fa-comment neutral-color"
                                        nifiTooltip
                                        [tooltipComponentType]="TextTip"
                                        [tooltipInputData]="rule.comments"></i>
                                }
                            </div>
                        </div>
                    }
                </div>
```